### PR TITLE
Fix deletion of old snapshots

### PR DESCRIPTION
**Pull request by codesome**
During last sanity check of snapshots I found a bug in deletion where the logic for selecting snapshot for deletion was not right. In this PR I am fixing it with some more enhancements.

1. Changed deletion logic from `idx <= maxIndex && offset < maxOffset` to `idx < maxIndex || (idx == maxIndex && offset < maxOffset)`
2. Instead of keeping last _2_ snapshots, I am now keeping only the last snapshot. Because the 2nd to last snapshot is basically useless.
3. Updated tests to catch this deletion bug.
4. Improved the selection of last checkpoint. Instead of relying on the file name sorting, now we check the numbers in the name.

A prerequisite for cutting 2.30 :)

### CODE REVIEW

File: tsdb/head_test.go [hunk 1]
```diff
@@ -2700,30 +2712,42 @@ func TestChunkSnapshot(t *testing.T) {
...
+	{
+		// Close Head and verify that new snapshot was created.
+		closeHeadAndCheckSnapshot()
+
+		// Verify that there is only 1 snapshot.
+		files, err := ioutil.ReadDir(head.opts.ChunkDirRoot)
 		require.NoError(t, err)
-		require.NoError(t, head.Init(math.MinInt64))
+		snapshots := 0
+		for i := len(files) - 1; i >= 0; i-- {
```
Comment (roidelapluie) [inline tsdb/head_test.go:2737]: That for is not really needed but potentially future-proof

**Review by roidelapluie — APPROVED** [review]
LGTM

File: tsdb/head_test.go (hunk 1 above)
Comment (codesome) [reply tsdb/head_test.go:2737]: There could be other directories, like wal, chunks_head. So we actually need this loop.

Comment (roidelapluie) [reply tsdb/head_test.go:2737]: ooh indeed

**Pull request merged**